include/rcutils/error_handling.h

```diff
@@ -103,6 +103,22 @@ rcutils_set_error_state(
 #define RCUTILS_SET_ERROR_MSG(msg, allocator) \
   rcutils_set_error_state(msg, __FILE__, __LINE__, allocator);
 
+/// Set the formatted error msg and free its memory allocated
+/**
+ * This function sets the error msg with the format specified and free the 
+ * corresponding memory allocated from function rcutils_format_string_limit() 
+ * or macro rcutils_format_string drived from it to avoid memory leak.
```

Comment:
I think that should be the responsibility of the person who calls `rcutils_format_string()` or `rcutils_format_string_limit()`.

Why do we need a function here to do that. What if someone uses `snprintf` or just did `malloc` and `memcpy` to create a message string? Should those get functions too? (I would say no)